### ISSUE TITLE
Dev server does not apply base path correctly

### ISSUE THREAD

**Issue by andycarlberg** [sveltejs/kit#2958]
### Describe the bug

I'm using a [code-server](https://github.com/cdr/code-server) cloud environment to develop a SvelteKit project. In order to access the dev server, a proxy is provided that uses a subpath to proxy the dev server port (see [docs](https://coder.com/docs/code-server/latest/guide#using-a-subpath). This leads me to need a base path to be set for the dev server. 

If I set `kit.paths.base` as stated in the documentation, the first page load works correctly but the `import` statements are relative to the root which returns a 404, correctly, because those files are not available without the proxy. If I run `build` and `preview` everything works as expected since the base path is built into the compiled project.

**Expected behavior**
All paths in the dev server reference the base path instead of being relative to the root domain.

### Reproduction

1. Create a new SvelteKit project
2. Set `kit.paths.base` to a non-root path
3. Run the dev server

**Expected result**
All `import` statements look for paths with the base path or relative to the base path

**Actual result**
Module `import` statements are relative to the root

**Example**
![image](https://user-images.githubusercontent.com/7405933/144281269-72b6ae18-a7ca-4e51-a684-f11c187fadc4.png)
![image](https://user-images.githubusercontent.com/7405933/144281372-087c2a6e-453a-4a5f-8ff7-6ac9c43c477c.png)

### Logs

```shell
GEThttp://domain.tld/.svelte-kit/dev/runtime/internal/start.js
[HTTP/1.1 404 Not Found 8ms]

Loading module from “http://domain.tld/.svelte-kit/dev/runtime/internal/start.js” was blocked because of a disallowed MIME type (“application/json”). 3000
Loading failed for the module with source “http://domain.tld/.svelte-kit/dev/runtime/internal/start.js”. 3000:13:1
GEThttp://domain.tld/favicon.png
[HTTP/1.1 404 Not Found 6ms]
```

### System Info

```shell
System:
    OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (12) x64 Intel(R) Core(TM) i5-10400 CPU @ 2.90GHz
    Memory: 283.55 MB / 15.32 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 14.18.2 - /usr/bin/node
    Yarn: 1.22.15 - /usr/bin/yarn
    npm: 6.14.15 - /usr/bin/npm
  npmPackages:
    @sveltejs/kit: ^1.0.0-next.201 => 1.0.0-next.201 
    svelte: ^3.42.6 => 3.44.2
```

### Severity

annoyance

### Additional Information

I've been using the `absproxy` method with code-server. This passes the full path with the `/absproxy/3000` portion intact. This seems like the correct approach for usage with `kit.paths.base`.  I think this method would work fine if the dev server accounted for it when building the pages.

However, there is also the `proxy` method. The `proxy` method strips of the proxy subpath before passing the request to the dev server. Thus, on the dev server, all the paths are relative to root. If there was a way to specify a subpath only for use on the client side, this may also be a reasonable solution.

I've made some inline edits to the `@sveltejs/kit` package in `node_modules` and was able to get the `proxy` method working. It would require a config field to set it, of course. Obviously, not everyone is working in a cloud server 😄

**Comment by andycarlberg** [comment]
Decided to see if I could fix this myself. After digging in it looks like this might actually be a Vite issue, not a SvelteKit issue.

vitejs/vite#5656

It's possible that SvelteKit may need additional updates beyond the fix for Vite as well.

**Comment by Bastian** [comment]
I'm having the same issue. My dev server is running behind a Traeffik Reverse Proxy and thus it fails to resolve the `.svelte-kit/*` files which are ignoring the `kit.paths.base` setting:
![image](https://user-images.githubusercontent.com/5033001/145988646-07c66e55-c1fe-4da4-8284-dc29f78e4fc6.png)

This is the config with the `kit.paths.base` set to `/demo`:
```ts
import adapter from '@sveltejs/adapter-node';
import preprocess from 'svelte-preprocess';

/** @type {import('@sveltejs/kit').Config} */
const config = {
	// Consult https://github.com/sveltejs/svelte-preprocess
	// for more information about preprocessors
	preprocess: [
		preprocess({
			postcss: true
		})
	],

	kit: {
		adapter: adapter(),

		// hydrate the <div id="svelte"> element in src/app.html
		target: '#svelte',
		ssr: true,
		trailingSlash: 'ignore',
		paths: {
			base: '/demo'
		}
	}
};

export default config;
```

**Comment by andycarlberg** [comment]
@Bastian is correct, this is not actually a Vite issue as I first suspected. It does seem to be specific to SvelteKit. So I'll keep looking but I'm probably out of my depth here for a quick fix.

I started looking at it as a Vite problem because I always seemed to come back to Vite when trying to trace the issue through SvelteKit. I must have missed something though since a vanilla Vite project applies the base path as expected.

**Comment by rassie** [comment]
@andycarlberg 
> I've made some inline edits to the @sveltejs/kit package in node_modules and was able to get the proxy method working. It would require a config field to set it, of course. Obviously, not everyone is working in a cloud server smile

Could you share a gist of what you've changed?

**Comment by andycarlberg** [comment]
@rassie 
I've pushed up some of my work-in-progress tinkering with the change essentially [here](https://github.com/andycarlberg/kit/blob/825552f613ca2bbf0519a74c15efa199c20144af/packages/kit/src/core/dev/index.js#L381).

This applies the base path prefix to the entry file. This is not a complete fix because I'm having trouble finding where these paths should actually be set. It's not a complete solution, more proof that there's an issue to be addressed I think.

**Comment by andycarlberg** [comment]
I feel like I'm on the right track here with this [change](https://github.com/andycarlberg/kit/blob/095e3c00be27a8386eb0de8bbdb06770e591d630/packages/kit/src/core/dev/index.js#L395).

It correctly outputs the file path with the base prefix but it 404's when looking for that `start.js` file. In debugging, it seems like that file is treated differently than other requests since the SvelteKit middleware doesn't process that file without my change (i.e. root-relative).

Any hints on where I might look to get it to recognize that file as the special start file even if it has the prefix?

**Comment by bjaskj** [comment]
I spent some time on something I think is similar, but I did not manage to figure out a fix for it.

I'm trying to use svelte kit via proxy server, and it works fine with static build and serve via other folder than root (having `kit.paths.base`) but it goes around and tries to include files that are in "root", like node modules etc. I only got it working if I set the base in vite config in the kit package and also modified the prefix, see below:

I think it boils down to the [core/dev/index.js](https://github.com/sveltejs/kit/blob/master/packages/kit/src/core/dev/index.js#L82) having a hardcoded value of "/", I tried playing around with this with some success but I ran into other issues trying to find a way to set it.

I also looked into `prefix` set to value of "" in this [core/dev/plugin.js](https://github.com/sveltejs/kit/blob/master/packages/kit/src/core/dev/plugin.js#L214)

This makes the paths to the script resources etc used in start method inside [runtime/server/page/render.js](https://github.com/sveltejs/kit/blob/master/packages/kit/src/runtime/server/page/render.js#L143)

Maybe someone can help with a fix on this?

**Comment by johnnysprinkles** [comment]
Not sure if this stackoverflow question is related: https://stackoverflow.com/questions/69378392/sveltekit-base-url-for-subdirectory-throws-404 They don't seem to be using a proxy though. I am using a proxy with stunnel and running into this issue, just working around with conditional svelteconfig, like:

    {
      kit: {
        paths: {
          base: process.env.NODE_ENV === 'development' ? undefined : '/mybase',
        },
      },
    }

**Comment by johnnysprinkles** [comment]
So it turns out this is only the index route that fails to load, any deeper path works fine.

**Comment by bjaskj** [comment]
What happen to me was that if I set kit.paths.base and then also the vite base path, which is hardcoded to '/' in kit, then I have to use `/mypath/mypath` to get to index. So something is a bit bork. Also if I managed to get it use `/mypath`, then all the other paths (node_modules, svelte kit etc) would still go to root/hostname `http://localhost/something`.

**Comment by johnnysprinkles** [comment]
So I just wanted to clarify and provide simple repo steps, and make sure I'm actually in the right bug (I can open a new one if not.) This doesn't have anything to do with proxies or missing js/css assets, it's actually that the main HTML document is 404 when using a base path and running in dev mode.

1. Init a brand new SvelteKit app with `npm init svelte@next sv` which gives as of now, SvelteKit v1.0.0-next.230
2. Choose the Demo App
3. cd sv; npm i
4. npm run dev
5. Everything works fine
6. Kill that, edit svelte.config.js to include `kit.paths.base = '/foo'`
7. Start up again with `npm run dev`
8. Note the both http://localhost:3000 and http://localhost:3000/foo give you a 404.
9. Note that deeper paths do work however, i.e. http://localhost:3000/foo/about

**Comment by benmccann** [comment]
> Note that both http://localhost:3000 and http://localhost:3000/foo give you a 404.

http://localhost:3000/foo works for me. Closing as fixed

**Comment by rassie** [comment]
@benmccann pardon me asking, maybe I'm missing something here, but what exactly has been fixed? You basically just said "works for me" (which is fine and probably true for the example above), but has there been any change addressing the original problem and the comments above?

**Comment by bjaskj** [comment]
I would also like to see this fixed, I can't use a dev env for some projects because of this issue.

**Comment by johnnysprinkles** [comment]
Thanks @benmccann I just tried with SvelteKit v1.0.0-next.236 and confirmed it's working. And to make sure I wasn't just doing something wrong before, I downgraded sveltekit to next.230 and observed the 404 again, so it seems to have been fixed.

There is a minor issue where the Demo app isn't prepending its static asset paths with `$app/paths.base`, so there's some broken images. I might file a separate super low priority bug for that.

It looks like I did hijack OPs original issue, sorry about that. Reading through it sounds like the want to slightly expand the meaning of `kit.paths.base` to skip over the base path in all contexts, even in local development. I've never heard of this "non-local local dev" before, I wonder how common it is?

**Comment by johnnysprinkles** [comment]
Also, sorry @benmccann if I was kind of a jerk last year, I was pulling my hair out trying to make SvelteKit a workable option for a very large project at Google, but that never did go anywhere and I ended up quitting Google. Onward!

**Comment by rassie** [comment]
@johnnysprinkles sorry to be blunt, and I don't think you intended this, but I think you derailed this issue for an unrelated problem.

@benmccann the problem in question still exists and it's a different one. In the example above, `localhost:3000/foo` is working, but the reason *why* it's working is because `localhost:3000/.svelte-kit` is still available without a proxy. If you put a proxy in front and only forward `/foo` to the server, you'd only forward requests to `/foo`, but not `/.svelte-kit`, you'll be getting 404 errors for SvelteKit runtime and nothing will work at all. The issue here is that `kit.paths.base` should be applied to `/.svelte-kit` too, which it's currently not.

You can easily verify it with the repro above, but look into Developer Tools, where you should see something like this:

![image](https://user-images.githubusercontent.com/23448/150510720-0f371394-952e-4c79-bc11-d6313d9a723b.png)

If `/.svelte-kit` is not properly forwarded as `/foo/.svelte-kit`, nothing will work. And forwarding just `/.svelte-kit` is not a solution, since the whole reason people use proxies is to host multiple sites on the same domain, so most people need a `/foo` with `/foo/.svelte-kit` and a `/bar` with `/bar/.svelte-kit`.

**Comment by andycarlberg** [comment]
As OP, i wanted to verify that @rassie is correct. Their comment shows the issue I was originally trying to deal with. I haven't personally made any further progress on it but would agree the issue is not fixed.

**Comment by tecoad** [comment]
> @johnnysprinkles sorry to be blunt, and I don't think you intended this, but I think you derailed this issue for an unrelated problem.
> 
> @benmccann the problem in question still exists and it's a different one. In the example above, `localhost:3000/foo` is working, but the reason _why_ it's working is because `localhost:3000/.svelte-kit` is still available without a proxy. If you put a proxy in front and only forward `/foo` to the server, you'd only forward requests to `/foo`, but not `/.svelte-kit`, you'll be getting 404 errors for SvelteKit runtime and nothing will work at all. The issue here is that `kit.paths.base` should be applied to `/.svelte-kit` too, which it's currently not.
> 
> You can easily verify it with the repro above, but look into Developer Tools, where you should see something like this:
> 
> ![image](https://user-images.githubusercontent.com/23448/150510720-0f371394-952e-4c79-bc11-d6313d9a723b.png)
> 
> If `/.svelte-kit` is not properly forwarded as `/foo/.svelte-kit`, nothing will work. And forwarding just `/.svelte-kit` is not a solution, since the whole reason people use proxies is to host multiple sites on the same domain, so most people need a `/foo` with `/foo/.svelte-kit` and a `/bar` with `/bar/.svelte-kit`.

I confirm this issue, I was meant to user sveltekit on a /blog subdirectory, however all assets still load from root, when it should rewrite to kit.paths.base. Unfortunately will have to stick with nextjs for this project.

**Comment by Beiri22** [comment]
Might be the same problem in my situation: I try to access the dev server via a reverse proxy. I have tried kit.paths.base and kit.vite.server.origin. Nothing works to set the sub-path associated on the proxy machine. Still showing ```import { start } from "/@fs/``` which should use some subdirectory...

**Comment by mosherbrian** [comment]
I am running a svelte-kit dev server behind a code-server proxy and can confirm that this is still an issue. Any fix or workaround available?

**Comment by AlbertMarashi** [comment]
I am still getting this issue. Any fixes?

**Comment by squelix** [comment]
I am still getting this issue. Any fixes?

**Comment by Rich-Harris** [comment]
Pretty sure this is fixed now — are people still experiencing it with latest versions? If so, a repro would be much appreciated

**Comment by Beiri22** [comment]
I'll try out on my example project as soon as possible.

**Comment by Beiri22** [comment]
@Rich-Harris The normal case works. But we have a special case here. The application is working with Shibboleth authentication that is carried out by an reverse proxy server. But all the "/@fs/..."-Links are not put under the base path. So I cannot use the dev mode. Is there any way to approach this behaviour?

By the way - in reverse proxy mode it would be nice to have an option that the base path is only applied when creating the pages. What do I mean: The page has a link to "/base/path/some.page", the reverse proxy removes the base path, so the server serves the document without the base path "/some.page"

**Comment by andycarlberg** [comment]
I just tried it out on my project with `svelte-kit, 1.0.0-next.335`

I'm still having the problem. I think @Beiri22 is correct, the `/@fs/...` paths don't have the base path so it's unable to load those supporting files in dev mode.

**Comment by andycarlberg** [comment]
1. Create a new SvelteKit project
2. Set kit.paths.base to a non-root path
3. Run the dev server

If you're just running the dev server locally, you _will_ be able to navigate the project. Check the browser dev tools and note the paths that are being requested. The `/@fs/...` paths don't include the base path prefix. This is the actual issue that needs to be addressed.

It works for local development because the local dev server has access to both the base path and the non-prefixed paths. It becomes a problem for development behind a proxy where the base path is necessary to access _any_ files on the dev server. In my case, this is surfaced when I'm trying to use [code-server](https://github.com/coder/code-server) to develop. The dev server isn't accessible locally because it isn't running locally. Instead, I access it through a proxied base path. It works for the initial page load but those `/@fs/...` paths can't be reached without the base path prefix.

**Comment by Beiri22** [comment]
I not 100% sure, but could it be that the /@fs/-paths are generated by vite rather than sveltekit?

**Comment by rassie** [comment]
`/@fs/` paths aren't the only ones, there is also `/.svelte-kit/`, which I'm pretty sure does not come from Vite ;)

**Comment by till** [comment]
> 1. Create a new SvelteKit project
> 2. Set kit.paths.base to a non-root path
> 3. Run the dev server
> 
> If you're just running the dev server locally, you _will_ be able to navigate the project. Check the browser dev tools and note the paths that are being requested. The `/@fs/...` paths don't include the base path prefix. This is the actual issue that needs to be addressed.
> 
> It works for local development because the local dev server has access to both the base path and the non-prefixed paths. It becomes a problem for development behind a proxy where the base path is necessary to access _any_ files on the dev server. In my case, this is surfaced when I'm trying to use [code-server](https://github.com/coder/code-server) to develop. The dev server isn't accessible locally because it isn't running locally. Instead, I access it through a proxied base path. It works for the initial page load but those `/@fs/...` paths can't be reached without the base path prefix.

I can confirm this. I did a `npm update` just now.

What works is: http://localhost:port/my-prefix

What doesn't work is when it sits behind a proxy: http://fancy.host.127.0.0.1.nip.io/my-prefix

Console shows a `404` in `start.ts`:

```
http://fancy.host.127.0.0.1.nip.io/@fs/Users/till/workspaces/app/src/.svelte-kit/runtime/client/start.js
```

**Comment by Beiri22** [comment]
@Rich-Harris This issue might not seem relevant, but for me it is. I cannot use the development mode in conjunction with the shibboleth authentication carried out by the reverse proxy server. This slows down the development...

**Comment by Beiri22** [comment]
It's really annoying. I need to run the programm behind a reverse proxy in order get the correct authentication information. I would run "dev" mode, but that does not work in that setup. So i need to run buikd and preview after each change in code.

**Comment by till** [comment]
> It's really annoying. I need to run the programm behind a reverse proxy in order get the correct authentication information. I would run "dev" mode, but that does not work in that setup. So i need to run buikd and preview after each change in code. 

Above 👆 there are the routes you need to add to your proxy. Posted as an example for nginx. Adapt, and it works. Do you need any help with another proxy?

**Comment by Beiri22** [comment]
I do only have access to a sub path, root @fs is no option...

**Comment by rassie** [comment]
> > It's really annoying. I need to run the programm behind a reverse proxy in order get the correct authentication information. I would run "dev" mode, but that does not work in that setup. So i need to run buikd and preview after each change in code.
> 
> Above point_up_2 there are the routes you need to add to your proxy. Posted as an example for nginx. Adapt, and it works. Do you need any help with another proxy?

Only works for one proxied site. Two or more is currently impossible.

**Comment by Beiri22** [comment]
In a setup with a reverse proxy, the "external base path" might or might not be preserved when forwarded to the server itself. So it we be nice to have either
a) thw base paths: an external that is used when creating all the links and an internal that is used when interpreting the requests
b) a flag that allows for the fact that the proxy might remove the external base path when relaying the request to the server.

**Comment by elron** [comment]
This issue still exists.

When I add this kit.paths
```
	const base = '/wp-content/plugins/lettering-product-shortcode-plugin';
	config.kit.paths = {
		base,
		assets: 'https://staging.hafonton.co.il' + base
	};
```

It totally ignores it and tries to load from root domain (/_app/immutable/assets/index-f01e1744.css)
instead of https://staging.hafonton.co.il/wp-content/plugins/lettering-product-shortcode-plugin/_app/immutable/assets/index-f01e1744.css

And it shows this error message:

`
Unable to preload CSS for /_app/immutable/assets/index-f01e1744.css
Error: Unable to preload CSS for /_app/immutable/assets/index-f01e1744.css
    at HTMLLinkElement.<anonymous> (https://staging.hafonton.co.il/wp-content/plugins/lettering-product-shortcode-plugin/_app/immutable/start-3d8db190.js:1:12256)
`

**Comment by JohnRSim** [comment]
> 1. Create a new SvelteKit project
> 2. Set kit.paths.base to a non-root path
> 3. Run the dev server
> 
> If you're just running the dev server locally, you _will_ be able to navigate the project. Check the browser dev tools and note the paths that are being requested. The `/@fs/...` paths don't include the base path prefix. This is the actual issue that needs to be addressed.
> 
> It works for local development because the local dev server has access to both the base path and the non-prefixed paths. It becomes a problem for development behind a proxy where the base path is necessary to access _any_ files on the dev server. In my case, this is surfaced when I'm trying to use [code-server](https://github.com/coder/code-server) to develop. The dev server isn't accessible locally because it isn't running locally. Instead, I access it through a proxied base path. It works for the initial page load but those `/@fs/...` paths can't be reached without the base path prefix.

Has this issue come back - just did a fresh install and set base path..

![image](https://user-images.githubusercontent.com/364208/190725655-0730c19f-d4ac-4efc-9755-97a59af5f23f.png)

```
/** @type {import('@sveltejs/kit').Config} */
import adapter from '@sveltejs/adapter-static';
import { mdsvex } from 'mdsvex';
import preprocess from 'svelte-preprocess';

const config = {
	extensions: ['.svelte', '.md', '.svelte.md'],
	preprocess: [
		mdsvex({ extensions: ['.svelte.md', '.md', '.svx'] }),
		preprocess({
			postcss: true,
			scss: {
				prependData: "@import 'src/lib/styles/variables.scss';",
			},
		}),
	],
	kit: {
		adapter: adapter({ precompress: true }),
		paths: {
			base: '/_themes/_components/BB-Custom-Form/publish/assets',
		},
	},
};

export default config;
```

**Comment by JohnRSim** [comment]
Ahh sorry was access through proxy - accessing direct works..

**Comment by BenediktAllendorf** [comment]
I didn't find a working configuration, but I've managed to patch the source code to work as desired.

I think there are two parts to this:
1. The import of [start.js](https://github.com/sveltejs/kit/blob/master/packages/kit/src/runtime/server/page/render.js#L225) must use the correct (base-)prefix.
2. The base for Vite must be set correctly [instead of the hardcoded `./`.](https://github.com/sveltejs/kit/blob/master/packages/kit/src/exports/vite/index.js#L243)

It worked as expected when adding my prefix ( `web`) into those two lines. I could run my SvelteKit-project at `localhost:3000/` and access it via my proxy at `localhost:9999/web`. Please note that I didn't add `paths.base` to `svelte.config.js` because then I would have had `localhost:3000/web/web` and `localhost:9999:/web/web` ( I guess that's an artifact of messing around).

I feel SvelteKit's understanding/usage of "base" differs from Vite's - how is it supposed to work here? I'm willing to create a PR for this, but I'd need some guidance (as in, is this even reasonable? How should this be tested?).

**Comment by benmccann** [comment]
I tried passing SvelteKit's base path to Vite awhile back, but found out that it's not currently possible because Vite handles the base path and static file serving in some ways that are different. We would need to resolve the following issues in Vite to make it possible to handle this correctly in SvelteKit:
- https://github.com/vitejs/vite/issues/9234
- https://github.com/vitejs/vite/issues/9260
- https://github.com/vitejs/vite/issues/9236

I would very much welcome PRs for those issues in the Vite repo. ViteConf is today. Vite 3.2 will be released later this week I believe and then work on Vite 4 can begin, which may enable us to make these changes, which would probably all be breaking changes. I can help check with the Vite team to ensure they're open to making these changes and agree on an approach, but would love help with implementing them.

**Comment by BenediktAllendorf** [comment]
I'm eager to help with the implementation (the alternative would be for us to maintain a fork and I'd rather see this being fixed upstream), but I'm cannot really asses right now what/where exactly should be changed. So if you, @benmccann, check with the Vite team, that would be great!

**Comment by benmccann** [comment]
I got a preliminary okay to address one of the issues and sent https://github.com/vitejs/vite/pull/10475 for it. However, I encountered what appears to be a Vite bug while fixing that. If you're able, I'd love to get a fix for that bug, which is something we could do now without further approval from the Vite team. See the TODO I'm adding here: https://github.com/vitejs/vite/pull/10475/files#diff-35ba301b85014a4bfaa9cad2d8e7eafa41c4e8c2ddd5c193182241d9a1542082R128

The Vite team has a meeting on Friday, so I'll try to get the other base path issues on the agenda for then

**Comment by benmccann** [comment]
A big thank you for figuring out the TODO @BenediktAllendorf !

I got the base path items on the agenda for Friday, so hopefully we'll come up with a plan then: https://github.com/orgs/vitejs/projects/1/views/8

**Comment by benmccann** [comment]
I talked to the Vite team this morning and made quite a bit of progress on a plan for moving forward. I sent https://github.com/sveltejs/kit/pull/7343 to address one of the items. I also sent https://github.com/vitejs/vite/pull/10590 to address a good chunk of https://github.com/vitejs/vite/issues/9236. @BenediktAllendorf I'll ping you after https://github.com/vitejs/vite/pull/10590 gets merged and perhaps if you're still interested you can finish the rest of https://github.com/vitejs/vite/issues/9236 which would be the last blocker.

**Comment by benmccann** [comment]
https://github.com/vitejs/vite/pull/10590 has been merged now. @BenediktAllendorf let me know if you'd be able to take a look at https://github.com/vitejs/vite/issues/9236. If we can get that fixed it should be the last blocker to setting the base path in Vite and removing our file serving in favor of Vite's. I updated that ticket with some details to help with getting started.

**Comment by rassie** [comment]
Seems like those issues have been fixed. Is anything else necessary to be able to say that Vite 4 will solve this?

**Comment by benmccann** [comment]
There is a Vite 4 branch of SvelteKit: https://github.com/sveltejs/kit/pull/7543. I'd appreciate if folks could test against it and let me know if it is fixed or not

**Comment by alexbjorlig** [comment]
I'm trying to figure out [why I can't serve Sveltekit from a different directory](https://github.com/sveltejs/kit/discussions/7717) - is this issue the culprit?

**Comment by benmccann** [comment]
This is no longer blocked by Vite, but I believe an additional change will be needed on our side

**Comment by duruer** [comment]
> This is no longer blocked by Vite, but I believe an additional change will be needed on our side

Yes, Vite fixed the issue, but seems like SvelteKit is still using `/node_modules` path.

**Comment by duruer** [comment]
Seems like one of the causing issue is this line:
https://github.com/sveltejs/kit/blob/master/packages/kit/src/exports/vite/dev/index.js#L96

If I change the line like this:
```
file: `/test${runtime_base}/client/start.js`,
```

It kinda fixes the issue but, then when you visit `/test`, it throws this error in console and shows 404 page in browser:
```
TypeError: Cannot create proxy with a non-object as target or handler
    at Module.handle_error_and_jsonify (/node_modules/@sveltejs/kit/src/runtime/server/utils.js:123:12)
    at Module.render_page (/node_modules/@sveltejs/kit/src/runtime/server/page/index.js:251:25)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async resolve (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:358:17)
    at async Module.respond (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:229:20)
    at async file:///home/kahverengi/Projects/Pano/panel-ui/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:468:22
```

**Comment by benmccann** [comment]
Perhaps this line:

https://github.com/sveltejs/kit/blob/06a56ae587795113b17fec559ab49c93f40861e8/packages/kit/src/runtime/server/page/render.js#L168

Needs to be changed to:
```js
const prefixed = (path) => (path.startsWith('/') ? base + path : `${resolved_assets}/${path}`);
```

Would someone affected by this problem be able to test it out?

**Comment by TheRevenantStar** [comment]
Looks like this change works, I'm using coder's code-server and serving from some-domain/proxy/5173

The page is loading the start script from the correct location.
<img width="883" alt="image" src="https://user-images.githubusercontent.com/40784277/213871633-74d6901b-5ef0-4445-9a12-5b244b4c0c14.png">

**Comment by janabimustafa** [comment]
The solution appears to be inadequate in handling cases where paths.relative is enabled but paths.base is absent, which was implemented to enable flexible base paths.

If a reverse-proxy is created and directed to the site using a path like /site-pr-10, the app.js file will load correctly but the start.js file will not. Essentially, this solution disrupts the functionality described in issue #595 when used behind a reverse proxy.
